zrml/swaps/src/lib.rs

```diff
@@ -1104,6 +1104,11 @@ mod pallet {
             amount: BalanceOf<T>,
         ) -> DispatchResult {
             let shares_id = Self::pool_shares_id(pool_id);
+            // Check that the account has at least as many free shares as we wish to burn!
+            ensure!(
+                T::Shares::ensure_can_withdraw(shares_id, from, amount).is_ok(),
+                Error::<T>::InsufficientBalance
+            );
```

Comment:
[nitpick] A little more rustacean way:
```suggestion
            T::Shares::ensure_can_withdraw(shares_id, from, amount)
                .map_err(|_| Error::<T>::InsufficientBalance)?;
```

Reply:
Perfect, thanks! 👍